smart_importer/predict_payees.py

```diff
@@ -131,34 +131,43 @@ def enhance_transactions(self):# load training data
         if not self._trained:
             logger.warning("Cannot generate predictions or suggestions "
                            "because there is no trained machine learning model.")
-            return self.imported_transactions
+            return self.imported_entries
+
+        updated_transactions: List[Transaction]
+        updated_transactions = list(filter_txns(self.imported_entries))
 
         # predict payees
-        self.transactions = self.imported_transactions
         if self.predict_payees:
             logger.debug("About to generate predictions for payees...")
             predicted_payees: List[str]
-            predicted_payees = self.pipeline.predict(self.transactions)
-            self.transactions = [ml.add_payee_to_transaction(*t_p, overwrite=self.overwrite_existing_payees)
-                            for t_p in zip(self.transactions, predicted_payees)]
+            predicted_payees = self.pipeline.predict(updated_transactions)
+            updated_transactions = [ml.add_payee_to_transaction(*t_p, overwrite=self.overwrite_existing_payees)
```

Comment:
Wording: Why "updated"? - I think I would rather name it `imported_transactions`.

* We start with a list of `imported_entries`. 
* When filtered to only include transactions, the list would then be called `imported_transactions`.
* Once predictions or suggestions have been added, we could call the list `enhanced_transactions`.

Note: I think I would prefer to have separate names for these lists, as opposed to using just one single name e.g. `transactions`. One reason why is that this would make the refactoring from #45 easier.